data/hai/hai reveal 3 early.txt

```diff
@@ -0,0 +1,429 @@
+			`	Content that the station is built satisfactorily, you prepare to return to <destination>.`
+				accept
+
+
+event "role-play station before construction"
+	system "Ultima Thule"
+		object
+			sprite star/f0
+			period 10
+		object
+			sprite planet/desert5
+			distance 200
+			period 22.3607
+		object
+			sprite planet/europa-b
+			distance 403.41
+			period 64.056
+		object
+			sprite planet/desert7-b
+			distance 824.9
+			period 187.302
+		object
+			sprite planet/gas1
+			distance 1706.99
+			period 557.553
+			object
+				sprite planet/ice8-b
+				distance 216
+				period 11.1964
+		object "Wormhole Alpha"
+			sprite planet/wormhole
+			distance 4208.63
+			period 2158.5
+			object Mountaintop
+				sprite "effect/flotsam yottrite"
+				distance 315
+				period 31.1964
+	planet Mountaintop
+		government "Hai"
+		attributes uninhabited
+		landscape land/station0
+		description `There's a construction site rapidly assembling a station from Hai Geocoris' orbiting this wormhole.`
+		"required reputation" -10
+		security 0.2
+
+
+event "role-play station after construction"
+	system "Ultima Thule"
+		object
+			sprite star/f0
+			period 10
+		object
+			sprite planet/desert5
+			distance 200
+			period 22.3607
+		object
+			sprite planet/europa-b
+			distance 403.41
+			period 64.056
+		object
+			sprite planet/desert7-b
+			distance 824.9
+			period 187.302
+		object
+			sprite planet/gas1
+			distance 1706.99
+			period 557.553
+			object
+				sprite planet/ice8-b
+				distance 216
+				period 11.1964
+		object "Wormhole Alpha"
+			sprite planet/wormhole
+			distance 4208.63
+			period 2158.5
+			object Mountaintop
+				sprite "planet/station hai eight geocoris dim"
+				distance 315
+				period 31.1964
+	planet Mountaintop
+		government "Hai"
+		attributes uninhabited
+		landscape land/station0
+		description `This station is made of eight Geocoris ships dismantled and reassembled into a spherical structure with external storage and landing platforms. Hai workers are shuffling about the station, installing human amenities and vast holographic projection systems. Presently it looks much like a construction site, but it is rapidly approaching completion.`
+		security 0.2
+
+
+mission "Hai Building Mountaintop [2] Recruits"
+	name "Encourage recruits for Mountaintop"
+	description "Now that the station is going ahead, the Hai need to recruit people to play the part. Someone needs to sell it to them, and apparently that someone is you."
+	landing
+	source "Hai-home"
+	stopover "Makerplace"
+	stopover "Darkmetal"
+	stopover "Giverstone"
+	stopover "Stonebreak"
+	stopover "Cloudfire"
+	stopover "Snowfeather"
+	stopover "Icelake"
+	stopover "Newhome"
+	stopover "Frostmark"
+	stopover "Heartvalley"
+	stopover "Allhome"
+	stopover "Dustmaker"
+	stopover "Greenbloom"
+	stopover "Skyfarm"
+	stopover "Farwater"
+	stopover "Greenwater"
```

Comment:
This one is meant to make you slightly resentful of it. The Narratives in ES, while covering vastly different scenarios, have very similar themes. In this case you've been dragged into a diplomatic mess, albeit one with a slightly non-serious bent. It will encourage you to appreciate it more when you go back into doing other major quests with more seriousness but less dumb politics afterwards. :P

Reply:
To expand on this slightly, presumably the same models Ms Adrianopoulos (I imagine she'd have my bank accounts frozen if I presumed to call her Terry without an invitation) uses to determine the number of people needed on the station would also provide some insight into where recruiting could be most effective.

Reply:
Or, if you want to be less subtle about it, have `ZZ Hai Building Mountaintop [2a] Sales Job` keep track of how many times you've done this and get grumpier as you go.

`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop.`
`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. The routine is well-practiced now, but starting to get a little tiresome.`
`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. You struggle to maintain a cheerful and enthusiastic facade as you count down the minutes until this ordeal is over.`
`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Why did you ever agree to this interminable dog and pony show? Is it really worth all this just to save one tiny little galaxy?`
`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. You're pretty sure you've done this more times than there are habitable planets in Hai space. Did you lose track and repeat a world? Will this nightmare ever end?`

(That last one is for when you actually *have* done it more times than there are habitable planets in Hai space. The mission fires every time you land, even if you've been there before.)

Reply:
Ok, if i can work out how to do it _that_ sounds like an approach I will take.

Reply:
Would something like this work?
```
	on offer
		"hai dog and pony" ++
		conversation
			branch toomany
				"hai dog and pony" < 17
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. You're pretty sure you've done this more times than there are habitable planets in Hai space. Did you lose track and repeat a world? Will this nightmare ever end?`
				decline
			label toomany
			branch interminable
				"hai dog and pony" < 14
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Why did you ever agree to this interminable dog and pony show? Is it really worth all this just to save one tiny little galaxy?`
				decline
			label interminable
			[...]
			label tiresome
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop.`
				decline
```

I haven't tested it, but I'm pretty sure all the individual pieces work. The slightly odd structure lets us keep each part together as a self-contained unit, but the following should be equivalent and might be easier to understand:

```
	on offer
		"hai dog and pony" ++
		conversation
			branch toomany
				"hai dog and pony" >= 17
			branch interminable
				"hai dog and pony" >= 14
			[...]
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop.`
				decline
			label tiresome
			[...]
			label interminable
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Why did you ever agree to this interminable dog and pony show? Is it really worth all this just to save one tiny little galaxy?`
				decline
			label toomany
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. You're pretty sure you've done this more times than there are habitable planets in Hai space. Did you lose track and repeat a world? Will this nightmare ever end?`
				decline
```

Note that in both cases we increment the counter before the conversation, so it will always include the current mission.

Reply:
Oh I like this, I believe one of them will work. I'll see how I go getting it in to work. Do you have a snapshot from just before this that you might be able to use to test for me by any chance?  (or give me to to use to test)

Reply:
Long delay I know, but I've now implemented this suggestion I believe.